### PR TITLE
Fix support for Nextcloud 27

### PR DESCRIPTION
**Pull request by smarinier**

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -48,7 +49,13 @@ function actionHandler(file: Node, dir: string) {
       path: dir,
     });
   } else {
-    downloadUrl = getAbsolutePath(file.encodedSource);
+	// NC28 dependency : Node.encodedSource
+	if (!file.encodedSource) {
+		const { origin } = new URL(file.source)
+		downloadUrl = getAbsolutePath(origin + encodePath(file.source.slice(origin.length)))
```
Comment (devnoname120) [inline src/main.ts:55]: Let's just revert to the old behavior behavior for this case. I don't want to risk breaking some edge cases.

File: src/main.ts [hunk 2]
```diff
@@ -48,7 +49,13 @@ function actionHandler(file: Node, dir: string) {
...
+	} else {
+	    downloadUrl = getAbsolutePath(file.encodedSource);
+	}
```
Comment (devnoname120) [inline src/main.ts:58]: Please fix the indentation. The code uses spaces and you added tabs here.

File: src/main.ts [hunk 3]
```diff
@@ -48,7 +49,13 @@ function actionHandler(file: Node, dir: string) {
       path: dir,
     });
   } else {
-    downloadUrl = getAbsolutePath(file.encodedSource);
+	// NC28 dependency : Node.encodedSource
```
Comment (devnoname120) [inline src/main.ts:52]: Could you rephrase the comment to instead say that pre-NC28 doesn't have Node.encodedSource?
Here it makes it sound like `Node.encodedSource` is a dependency of NC28 and thus required by NC28 (it's not) which is confusing.

**Review by devnoname120 — CHANGES_REQUESTED** [review]
Thanks!

File: src/main.ts (hunk 1 above)
Comment (smarinier) [reply src/main.ts:55]: As you like. This was keeping the fix you've done in april.

**Review by devnoname120 — APPROVED** [review]
